node_modules/oae-core/editprofile/js/editprofile.js

```diff
@@ -103,6 +103,11 @@ define(['jquery', 'oae.core'], function ($, oae) {
             // has no real display name
             if (!isValidDisplayName()) {
```

Comment:
The `isValidDisplayName` validation function should have exactly the same checks as the back-end checks. It looks like it currently checks for `http(s)` and `@`, but not `shibboleth!`. Can you add that in as well?